main.rb

```diff
@@ -1 +1,32 @@
+def decode_char(char)
+  chars = {
+    '.-' => 'A', '-...' => 'B', '-.-.' => 'C', '-..' => 'D', '.' => 'E',
+    '..-.' => 'F', '--.' => 'G', '....' => 'H', '..' => 'I', '.---' => 'J',
+    '-.-' => 'K', '.-..' => 'L', '--' => 'M', '-.' => 'N', '---' => 'O',
+    '.--.' => 'P', '--.-' => 'Q', '.-.' => 'R', '...' => 'S', '-' => 'T',
+    '..-' => 'U', '...-' => 'V', '.--' => 'W', '-..-' => 'X', '-.--' => 'Y',
+    '--..' => 'Z'
+  }
+  chars.each do |key, value|
+    return value if key == char
+  end
+end
 
+def decode_word(coded_word)
+  actual_word = ''
+  coded_word.split.each do |value|
+    char = decode_char(value)
+    actual_word += char
+  end
+  actual_word
+end
+
+def decode_sentence(coded_sentence)
+  msg = []
+  coded_sentence.split('  ').each do |word|
+    msg.push(decode_word(word))
+  end
+  msg.join(' ')
+end
+
```

Comment:
When I run the following test code:

![image](https://user-images.githubusercontent.com/38088818/191520060-9fc58f22-0890-4461-b005-341baa9d15a3.png)

To get the expected result:

![image](https://user-images.githubusercontent.com/38088818/191520254-15f377c3-ed45-4d9a-99d9-28bf4182f32d.png)

I get the correct result for the most part, especially with the `decode_char` and `decode_word` methods. Well done!!! 👏🏾 
However, the `decode_sentence` method outputs sentences with two spaces between words.

![image](https://user-images.githubusercontent.com/38088818/191521224-6b3178ba-85ac-4094-9767-6fa03ba1751f.png)

It could be the way you split the `coded_sentence`:

![image](https://user-images.githubusercontent.com/38088818/191521532-76311834-7097-46d7-9b94-2d1f2db1d61c.png)

- Kindly look into it to make sure that the decoded message forms a grammatically correct sentence with exactly a single space between words.

Reply:
Hello @KDlamini, thanks for giving us feedback but the reason for spriting with 2 spaces is that even in sentence words are separated by two spaces